File: src/main/java/org/git/desk/component/PlatformComboBox.java
package org.git.desk.component;

import javafx.scene.control.ComboBox;
import org.git.desk.constant.GitPlatform;

public class PlatformComboBox extends ComboBox<GitPlatform> {

  public PlatformComboBox() {
    super();
    init();
  }

  private void init() {
    // 添加所有平台枚举到下拉列表
    this.getItems().addAll(GitPlatform.values());

    // 设置自定义 CellFactory
    PlatformCellFactory cellFactory = new PlatformCellFactory();
    this.setCellFactory(cellFactory);
    this.setButtonCell(cellFactory.call(null));

    // 默认选中第一个
    if (!this.getItems().isEmpty()) {
      this.getSelectionModel().selectFirst();
    }
  }
}

File: src/main/kotlin/org/git/desk/component/PlatformComboBox.kt
package org.git.desk.component

import javafx.scene.control.ComboBox
import org.git.desk.constant.GitPlatform

class PlatformComboBox : ComboBox<GitPlatform>() {
  init {
    init()
  }

  private fun init() {
    items.addAll(*GitPlatform.entries.toTypedArray())
    val cellFactory = PlatformCellFactory()
    this.cellFactory = cellFactory
    this.buttonCell = cellFactory.call(null)
    this.selectionModel.selectFirst()
  }
}